plugins/plugin-docker/che-plugin-docker-machine/src/test/java/org/eclipse/che/plugin/docker/machine/MachineProviderImplTest.java

```diff
@@ -697,6 +697,30 @@ public void shouldBindCommonAndDevVolumesToContainerOnDevInstanceCreationFromRec
         assertEquals(actualBinds.length, expectedVolumes.size());
         assertEquals(new HashSet<>(asList(actualBinds)), new HashSet<>(expectedVolumes));
     }
+    
+    @Test
+    public void shouldBindVolumesFromToContainerOnDevInstanceCreationFromRecipe() throws Exception {
+    	
+    	List<String> cheServerVolumes = new ArrayList<>(asList("SomeVolumeName:/data/lib"));
+    	String expectedVolumesFrom = "che-server";
+    	
+    	provider = new MachineProviderBuilder().setVolumesFrom(expectedVolumesFrom)
+    									       .build();
+    	
+    	final boolean isDev = true;
+    	
+    	CheServiceImpl service = createService();
+    	service.setContainerName(expectedVolumesFrom);
+    	service.setVolumes(cheServerVolumes);
+    	createInstanceFromRecipe(isDev, service);
+    	
+        ArgumentCaptor<CreateContainerParams> argumentCaptor = ArgumentCaptor.forClass(CreateContainerParams.class);
+        verify(dockerConnector).createContainer(argumentCaptor.capture());
+        verify(dockerConnector).startContainer(any(StartContainerParams.class));
+        
+        final String[] actualVolumesFrom = argumentCaptor.getValue().getContainerConfig().getHostConfig().getVolumesFrom();
+        assertTrue(Arrays.asList(actualVolumesFrom).contains(expectedVolumesFrom));
+    }
```

Comment:
Please add one more test with volumes from set in che service to ensure that combination of volumesfrom from che service and predefined volumesfrom don't wipe-out each other.